### ISSUE TITLE
CPU Useage on Dedicated Server

### ISSUE THREAD

**Issue by pjlosey** [alvyxaz/barebones-masterserver#30]
Im probably doing this the wrong way. 

I setup the demo and uploaded a copy to my dedicated windows server. It will run but maxes out my cpu once the server, and two worlds are running.   

I am assuming there is a way to run them in something like a console (Used to hosting rust, ark servers). Is this possible?

Also do you have guide on what i need to change to have the clients settings set to to get to the server from anywhere (ip's and ports). I have seen a few settings just not sure what im missing.

Thanks

**Comment by delmarle** [comment]
you should use appllication.targetframerate = 30;

**Comment by pjlosey** [comment]
> you should use appllication.targetframerate = 30;

That is set by default i belive. overhead is still way to large compared to say rust servers i have running 150-200 people using about 10% of my cpu at 250+ fps on the server only showing a console.  Seeing the demo files are very small im thinking there has to be some other way of hosting that im not seeing lol

**Comment by delmarle** [comment]
by default there is no target framerate so it try to run it as fast as it can, thats why it max out cpu usage

**Comment by pjlosey** [comment]
<img width="472" alt="30fps" src="https://cloud.githubusercontent.com/assets/105854/22181152/4c541678-e03a-11e6-9325-1e030a9402bd.PNG">

assuming this is the correct place for this setting, mine was at 30fps already.

**Comment by pjlosey** [comment]
<img width="471" alt="batchmode" src="https://cloud.githubusercontent.com/assets/105854/22181172/d4b23270-e03a-11e6-91a0-9d4ce87e0d58.PNG">

not sure if batchmode is what im looking for.

**Comment by delmarle** [comment]
yes you have to run it in batchmode for dedicated server, it will not load graphics on only run scripts

**Comment by pjlosey** [comment]
now for the better noob question. how do i run it in batch mode on windows. so far the ways i have tried still fire the graphics

**Comment by delmarle** [comment]
check https://docs.unity3d.com/Manual/CommandLineArguments.html

On macOS, type the following into the Terminal to silently launch Unity:

/Applications/Unity/Unity.app/Contents/MacOS/Unity -quit -batchmode -serial    R3-XXXX-XXXX-XXXX-XXXX-XXXX -username 'JoeBloggs@example.com' -password 'MyPassw0rd'
On 64-bit Windows, type the following into the Command Prompt to silently launch Unity:

     "C:\Program Files\Unity\Editor\Unity.exe" -quit -batchmode -serial    R3-XXXX-XXXX-XXXX-XXXX-XXXX -username "JoeBloggs@example.com" -password "MyPassw0rd"
On 32-bit Windows, type the following into the Command Prompt to silently launch Unity:

 "C:\Program Files (x86)\Unity\Editor\Unity.exe" -quit -batchmode -serial    R3-XXXX-XXXX-XXXX-XXXX-XXXX -username "JoeBloggs@example.com" -password "MyPassw0rd"

**Comment by pjlosey** [comment]
i belive the info above is for the unity editor right? 

If i run a shortcut to the build for the demo with the following paramaters it runs in batch mode,   C:\Users\pjlos\Desktop\Unity\StudioTwentyOne\_Builds\MasterServerFrameWorkTesting\build.exe -bmMaster -bmSpawner -spawnZones -batchmode


however the next problem i have is the world servers still open in normal mode and not in batchmode so they still show the graphics.  Any hints there?

**Comment by pjlosey** [comment]
in SpawnerServers.cs i modified line 412 to 

-batchmode " +

from 

(RunInBatchMode ? "-batchmode " : "") +

it would batchmode the worlds if it was ran in the editor but not though the exe.

hoping that means something lol

**Comment by pjlosey** [comment]
last part of my fun.   I have the server running on the dedicated box. the complete demo it all runs fine, runs in batch mode now after making a bat file to run it with the parameters.

my next issue is i go to join the world /zones demo from  client on my home pc (Different ip)  and i get connection to 127.0.0.1 and then kicked back to the previous menu.    i have changed all the ip's i could find to the remote ip address.

Any Suggestions.

**Comment by alvyxaz** [comment]
I'll look into the previous messages a bit later, as I'm away and don't
have much time, but to help you with your latest problem, check out this
page:

https://github.com/alvyxaz/barebones-masterserver/wiki/Hosting-Your-Game

A few days ago I've updated it with a warning about "connecting to
127.0.0.1"

Let me know if this fixes the issue, and sorry about all the inconveniences

**Comment by alvyxaz** [comment]
By the way, @delmarle  - thanks a lot for stepping in!

**Comment by pjlosey** [comment]
@alvyxaz  no problme at all. sorry for all the text chat :)  working it out is what we do lol.  i tried that on my bat file but its not working. modified the spawnerservers file as well as an option.  stuck here, crashing now ty for the help.  happy to share my project if you want to look at it :)

**Comment by pjlosey** [comment]
http://216.52.148.233/MasterServerFrameWorkTesting.zip

**Comment by alvyxaz** [comment]
Hey! So you're still having an issue where client is connecting to "127.0.0.1"? :)

**Comment by pjlosey** [comment]
i get one of two results. either i end up with the 127.xxxx or i hit a wall of the the world servers not existing, have to re do it to see the exact error again.

Im sure its just my lack of experience here but not sure. excited when you get a guide going of how to put a server on  another machine and client on a separate.
